src/pint/simulation.py

```diff
@@ -514,25 +514,14 @@ def make_fake_toas_fromtim(
     --------
     :func:`make_fake_toas`
     """
-    ephem = (
-        model.EPHEM.value
-        if hasattr(model, "EPHEM") and model.EPHEM.value is not None
-        else None
-    )
-    planets = (
-        model.PLANET_SHAPIRO.value
-        if hasattr(model, "PLANET_SHAPIRO") and model.PLANET_SHAPIRO.value is not None
-        else False
-    )
-
-    input_ts = pint.toa.get_TOAs(timfile, ephem=ephem, planets=planets)
+    input_ts = pint.toa.get_TOAs(timfile, model=model)
 
     if input_ts.is_wideband():
         dm_errors = input_ts.get_dm_errors()
         ts = update_fake_dms(model, input_ts, dm_errors, add_noise)
```

Comment:
You are creating `ts` here if it's Wideband, but if it's not Wideband then I don't know if that variable is used.  which makes the command below suspect.

Reply:
Corrected.